### PR TITLE
Samples, toolset for async

### PR DESCRIPTION
**Pull request by howieleung**
More samples.
Defined AsyncToolset and AsyncFunctionTool that can execute both sync or async functions.

### CODE REVIEW

File: sdk/ai/azure-ai-assistants/azure/ai/assistants/aio/_patch.py [hunk 1]
```diff
@@ -29,7 +28,212 @@
...
+        :paramtype response_format: ~azure.ai.assistants.models.AssistantsApiResponseFormatOption
+        :keyword metadata: Key/value pairs for storing additional information. Default value is None.
+        :paramtype metadata: dict[str, str]
+        :return: An Assistant object.
+        :rtype: ~azure.ai.assistants.models.Assistant
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+    @distributed_trace_async
+    async def create_assistant(
```
Comment (jhakulin) [inline sdk/ai/azure-ai-assistants/azure/ai/assistants/aio/_patch.py:171]: have you integrated create_and_process_run also?

File: sdk/ai/azure-ai-assistants/azure/ai/assistants/models/_patch.py [hunk 2]
```diff
@@ -355,6 +377,42 @@ def execute_tool_calls(self, tool_calls: List[Any]) -> Any:
...
+    def validate_tool_type(self, tool_type: Type[Tool]) -> None:
+        """
+        Validate the type of the tool.
+
+        :param tool_type: The type of the tool to validate.
+        :raises ValueError: If the tool type is not a subclass of Tool.
+        """
+        if isinstance(tool_type, FunctionTool):
+            raise ValueError("FunctionTool is not supported in AsyncToolSet.  Please use AsyncFunctionTool instead and provide sync and/or async function(s).")
+
```
Comment (jhakulin) [inline sdk/ai/azure-ai-assistants/azure/ai/assistants/models/_patch.py:391]: remove extra line

File: sdk/ai/azure-ai-assistants/samples/async_samples/sample_assistant_stream_eventhandler_async.py [hunk 3]
```diff
@@ -114,14 +114,13 @@ async def sample_assistant_stream_iteration():
         ) as stream:
             await stream.until_done()
 
-        messages = await assistant_client.list_messages(thread_id=thread.id)
-        logging.info(f"Messages: {messages}")
-
         await assistant_client.delete_assistant(assistant.id)
         logging.info("Deleted assistant")
 
+        messages = await assistant_client.list_messages(thread_id=thread.id)
```
Comment (jhakulin) [inline sdk/ai/azure-ai-assistants/samples/async_samples/sample_assistant_stream_eventhandler_async.py:120]: was there a reason to change the order? just curious

Comment (howieleung) [reply sdk/ai/azure-ai-assistants/samples/async_samples/sample_assistant_stream_eventhandler_async.py:120]: I found to read the message in terminal, I need to scroll up to skip lot of content.   So I move the message down.

File: sdk/ai/azure-ai-assistants/azure/ai/assistants/models/_patch.py (hunk 2 above)
Comment (howieleung) [reply sdk/ai/azure-ai-assistants/azure/ai/assistants/models/_patch.py]: Done

**Pull request merged**